### ISSUE TITLE
Server not listed. Server online: No

### ISSUE THREAD

**Issue by arfianadam** [arkmanager/ark-server-tools#866]
I have wandered everywhere and tried almost everything. Still, can not figure out why it is still not online on Steam Master server. Here is my `main.cfg`
```
# config environment
arkserverroot="/home/steam/ARK"                                     # path of your ARK server files (default ~/ARK)

# ARK server options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap="TheIsland"                                               # server map (default TheIsland)
#serverMapModId="469987622"                                         # Uncomment this to specify the Map Mod Id (<fileid> in http://steamcommunity.com/sharedfiles/filedetails/?id=<fileid>)
#ark_TotalConversionMod="496735411"                                 # Uncomment this to specify a total-conversion mod
ark_RCONEnabled="True"                                              # Enable RCON Protocol
ark_RCONPort="32330"                                                # RCON Port
ark_Port="7778"                                                     # ARK server port (default 7778)
ark_QueryPort="27015"                                               # ARK query port (default 27015)
ark_bRawSockets="True"
ark_ServerPassword=""                                               # ARK server password, empty: no password required to login
ark_ServerAdminPassword="***"                               # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers="70"
ark_GameModIds="878297372,558079412,700624475,731604991,821530042,928102085,693416678,895711211"                     # Uncomment to specify additional mods by Mod Id separated by commas
#ark_AltSaveDirectoryName="SotF"                                    # Uncomment to specify a different save directory name

# ARK server flags - use arkflag_<optionname>=true
#arkflag_OnlyAdminRejoinAsSpectator=true                            # Uncomment to only allow admins to rejoin as spectator
#arkflag_DisableDeathSpectator=true                                 # Uncomment to disable players from becoming spectators when they die

# ARK server options - i.e. for -optname=val, use arkopt_optname=val
#arkopt_StructureDestructionTag=DestroySwampSnowStructures
```

It is running and I can join directly using IP. I have checked that I have no firewall running and everything is open. I even used `nmap` to check open ports listed on `netstat`. They are all accessible.

---
`iptables -L`
```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:7777
ACCEPT     udp  --  anywhere             anywhere             udp dpt:7777
ACCEPT     udp  --  anywhere             anywhere             udp dpt:27015
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:32330
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:7778
ACCEPT     udp  --  anywhere             anywhere             udp dpt:7778
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:27016
ACCEPT     udp  --  anywhere             anywhere             udp dpt:27016

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
```
---
`netstat -tuplen`
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode       PID/Program name
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      0          42948       29924/sshd      
tcp        0      0 0.0.0.0:32330           0.0.0.0:*               LISTEN      1000       646190      31062/ShooterGameSe
tcp6       0      0 :::22                   :::*                    LISTEN      0          42950       29924/sshd      
udp        0      0 0.0.0.0:7777            0.0.0.0:*                           1000       644773      31062/ShooterGameSe
udp        0      0 0.0.0.0:7778            0.0.0.0:*                           1000       646310      31062/ShooterGameSe
udp        0      0 0.0.0.0:27015           0.0.0.0:*                           1000       642769      31062/ShooterGameSe
```
---
Anyone has any idea? My server ip is at `172.104.165.254`. Please do try to check the ports yourself if that helps. Thank you!

**Comment by klightspeed** [comment]
That looks like you're running two servers, one of which has only opened the game port.  Either that, or the ARK server is now binding to both the Port and GamePort (Port+1) ports.

The server is showing up in the steam master list:
http://api.steampowered.com/ISteamApps/GetServersAtAddress/v0001?addr=172.104.165.254
```
{
	"response": {
		"success": true,
		"servers": [
			{
				"addr": "172.104.165.254:27015",
				"gmsindex": 65534,
				"appid": 346110,
				"gamedir": "ark_survival_evolved",
				"region": -1,
				"secure": true,
				"lan": false,
				"gameport": 7777,
				"specport": 0
			}
		]
		
	}
}
```

It is also showing up on arkservers.net:
https://arkservers.net/server/172.104.165.254:27015

**Comment by arfianadam** [comment]
Ummm I'm sure I am running only one server. Can you please elaborate?

> The server is showing up in the steam master list:

> It is also showing up on arkservers.net:

Yes, that's what I checked, the server is available and validated. But still, when I search on steam servers, I can't find it.

**Comment by arfianadam** [comment]
Thanks for the reply @klightspeed. Ummm I'm sure I am running only one server. Can you please elaborate?

> The server is showing up in the steam master list:

> It is also showing up on arkservers.net:

Yes, that's what I checked, the server is available and validated. But still, when I search on steam servers, I can't find it.

**Comment by klightspeed** [comment]
I suspect there are so many servers out there that both the in-game server browser and the Steam server browser give up after listing about 10k-15k servers.

**Comment by arfianadam** [comment]
> I suspect there are so many servers out there that both the in-game server browser and the Steam server browser give up after listing about 10k-15k servers.
It's not even 10k right now, mine is on 4k and stopped haha

**Comment by cronner** [comment]
got the same problem :/

**Comment by klightspeed** [comment]
Does it work if you run the `ShooterGameServer` directly, using the options you have specified (written in `arkserver.log`)?

**Comment by lafaetos** [comment]
hello, I have a similar problem. In the configuration file I specified port 7777, but it try starts on port 7776.
> arkmanager status
> Running command 'status' for instance 'main'
>  Server running:   Yes 
>  Server PID:   2292 
>  Server listening:   Yes 
> Server Name: [UA]Arkanion|x10|PVE\PVP|Ragnarok|Castle|CFL|Structure+| - (v27
> Players: 0 / 20
> Active Players: 0
>  Server online:   No 
>  Server build ID:   2204510 

> ps aux| grep Shooter
steam     2292 22.9 47.5 7788716 5781700 ?     Sl   14:40  23:50 /steam/arkdedicated/ShooterGame/Binaries/Linux/ShooterGameServer Ragnarok?RCONEnabled=True?RCONPort=32330?SessionName=[UA]Arkanion|x10|PVE\PVP|Ragnarok|Castle|CFL|Structure+|?Port=7776?QueryPort=27015...

> netstat -tuplen
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode       PID/Program name
tcp        0      0 0.0.0.0:32330           0.0.0.0:*               LISTEN      1001       29597       2292/ShooterGameSer
udp        0      0 0.0.0.0:27015           0.0.0.0:*                           1001       22773       2292/ShooterGameSer
udp        0      0 0.0.0.0:7777            0.0.0.0:*                           1001       26599       2292/ShooterGameSer

> lsof -i -P -n| grep 777
> ShooterGa  2292    steam   18u  IPv4  26599      0t0  UDP *:7777 

and have another problem, i set next parametr:
> arkCpuAffinity=2
at startup, writes that the kernels have changed, but still works at all

if you apply taskset pid 2, it helps
thanks for the help

**Comment by dahlskebank** [comment]
~~Did anyone ever figure this out? Not that it looks like an issue, but I'm still getting ``Server online: No`` even though all ports are opened. Listed on arkservers.net, ark-servers.net, and battlemetrics.com, even on the steam API link. Can connect and play on the server, so not sure when this, if it ever will be an issue.~~

Hm. The issue went away when I updated arkmanager... I guess if everyone had the same issue as me, this could be marked as solved.

**Comment by JimmyMEG1** [comment]
I got the same problem and I notice that : 
Modification of /usr/local/bin/arkmanager

`function isTheServerOnline(){
  if [ -n "$(getMultiHome "")" ]; then
    **# comment this
    #publicip="$(curl --interface "$(getMultiHome)" -s http://api.ipify.org/)"
    # for this
    publicip="$(curl -s http://api.ipify.org/)"**
  else
    publicip="$(curl -s http://api.ipify.org/)"
  fi
`
After this change and after restart the server, the server is online on arkservers.net.

My getMultiHome is my real public IP address but my servers have private ip address behind a NAT.
In that way, curl cannot works properly.

**Comment by cronner** [comment]
@JimmyMEG1 This was the same case here, this seems to fix it

**Comment by klightspeed** [comment]
This was fixed in 6da2c49 (in v1.6.35)
